src/rebase.rs

```diff
@@ -0,0 +1,339 @@
+    pub rewrite_notes_ref: Option<CString>,
+
+    /// Options to control how trees are merged during a `rebase`.
+    pub merge_options: Option<MergeOptions>,
+
+    /// Options to control how files are written during a `rebase`.
+    pub checkout_builder: Option<CheckoutBuilder<'a>>,
+}
+
+impl<'repo> Rebase<'repo> {
+    /// Returns the number of applied operations.
+    pub fn operation_count(&self) -> usize {
+        unsafe { raw::git_rebase_operation_entrycount(self.raw) }
+    }
+
+    /// Aborts a rebase that is currently in progress,
+    /// resetting the repository and working directory to their state before rebase began.
+    pub fn abort(&self) -> Result<(), Error> {
+        unsafe {
+            try_call!(raw::git_rebase_abort(self.raw));
+        }
+        Ok(())
+    }
+
+    /// Commits the current patch.
+    /// You must have resolved any conflicts
+    /// that were introduced during the patch application from the `next` invocation.
+    pub fn commit(&self,
+                  author: Option<&Signature>,
+                  committer: &Signature,
+                  message: Option<&str>)
+                  -> Result<Oid, Error> {
+        let mut raw = raw::git_oid { id: [0; raw::GIT_OID_RAWSZ] };
+        let message = try!(::opt_cstr(message));
+        unsafe {
+            try_call!(raw::git_rebase_commit(&mut raw,
+                                             self.raw,
+                                             author.map(|s|s.raw() as *const _),
+                                             committer.raw(),
+                                             0 as *const c_char,
+                                             message));
+            Ok(Binding::from_raw(&raw as *const _))
+        }
+    }
+
+    /// Finishes a rebase that is currently in progress once all patches have been applied.
+    pub fn finish(&self, signature: Option<&Signature>) -> Result<(), Error> {
+        unsafe {
+            try_call!(raw::git_rebase_finish(self.raw,
+                                             signature
+                                                 .map(|s| {
+                                                          s.raw() as *const raw::git_signature
+                                                      })
+                                                 .unwrap_or(0 as *const raw::git_signature)));
+        }
+        Ok(())
+    }
+    /// Gets the index produced by the last operation.
+    /// This is useful for resolving conflicts in an in-memory rebase before committing them.
+    pub fn inmemory_index(&self) -> Result<Index, Error> {
+        let mut ret = 0 as *mut raw::git_index;
+        unsafe {
+            try_call!(raw::git_rebase_inmemory_index(&mut ret, self.raw));
+            Ok(Binding::from_raw(ret))
+        }
+    }
+
+    /// Performs the next rebase operation and returns the `RebaseOperation` about it.
+    pub fn next(&self) -> Result<RebaseOperation, Error> {
```

Comment:
I wonder if this may be best exposed as an `Iterator` structure perhaps? Maybe a manual iterator over a rebase entry?